pkg/api/v1alpha1/cluster_test.go

```diff
@@ -93,6 +93,52 @@ func TestClusterNameLength(t *testing.T) {
 	}
 }
 
+func TestValidateExternalEtcdSupport(t *testing.T) {
+	g := NewWithT(t)
+	tests := []struct {
+		name        string
+		wantCluster *Cluster
+		wantErr     bool
+	}{
+		{
+			name: "tinkerbell config without external etcd",
+			wantCluster: &Cluster{
+				Spec: ClusterSpec{
+					DatacenterRef: Ref{
+						Kind: TinkerbellDatacenterKind,
+						Name: "eksa-unit-test",
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "tinkerbell config with external etcd",
+			wantCluster: &Cluster{
+				Spec: ClusterSpec{
+					ExternalEtcdConfiguration: &ExternalEtcdConfiguration{Count: 1},
+					DatacenterRef: Ref{
+						Kind: TinkerbellDatacenterKind,
+						Name: "eksa-unit-test",
+					},
+				},
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(tt *testing.T) {
+			got := validateExternalEtcdSupport(tc.wantCluster)
+			if tc.wantErr {
+				g.Expect(got).To(MatchError(ContainSubstring("unsupported")))
```

Comment:
```suggestion
				g.Expect(got).To(MatchError(ContainSubstring("external etcd configuration is unsupported")))
```

What about using more of the string like this? I think this would be more robust since this test is looking for external etcd configuration support specifically